src/Orleans.Runtime/Transactions/TransactionAgent.cs

```diff
@@ -203,8 +205,10 @@ public TransactionAgent(ILocalSiloDetails siloDetails, ITransactionManagerServic
             await completion.Task;
         }
 
-        public void Abort(TransactionInfo transactionInfo, OrleansTransactionAbortedException reason)
+        public void Abort(ITransactionInfo info, OrleansTransactionAbortedException reason)
         {
+            var transactionInfo = (TransactionInfo)info;
```

Comment:
This violates strong behavioral subtyping, which we do in some cases, but never without reason.
Under what conditions would info not be a TransactionInfo?  Can we throw a more meaningful error in this case?

Reply:
I don't see any circumstance in which this could be the wrong type.